fdbserver/SimEncryptVaultProxy.actor.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * SimEncryptVaultProxy.actor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#if defined(NO_INTELLISENSE) && !defined(FDBSERVER_SIMENCRYPTVAULTPROXY_ACTOR_G_H)
+#define FDBSERVER_SIMENCRYPTVAULTPROXY_ACTOR_G_H
+#include "fdbserver/SimEncryptVaultProxy.actor.g.h"
+#elif !defined(FDBSERVER_SIMENCRYPTVAULTPROXY_ACTOR_H)
+#define FDBSERVER_SIMENCRYPTVAULTPROXY_ACTOR_H
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbrpc/fdbrpc.h"
+#include "flow/FileIdentifier.h"
+#include "flow/Trace.h"
+#include "flow/flow.h"
+#include "flow/network.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+using SimEncryptKeyId = uint64_t;
+using SimEncryptDomainId = uint64_t;
+using SimEncryptKey = std::string;
+
+struct SimEncryptVaultProxyInterface {
+	constexpr static FileIdentifier file_identifier = 2416711;
+	RequestStream<ReplyPromise<Void>> waitFailure;
+	RequestStream<struct SimGetEncryptKeyByKeyIdRequest> encryptKeyLookupByKeyId;
+	RequestStream<struct SimGetEncryptKeyByDomainIdRequest> encryptKeyLookupByDomainId;
+
+	SimEncryptVaultProxyInterface() {}
+
+	UID id() const { return encryptKeyLookupByKeyId.getEndpoint().token; }
+	template <class Archive>
+	void serialize(Archive& ar) {
+		if constexpr (!is_fb_function<Archive>) {
+			ASSERT(ar.protocolVersion().isValid());
+		}
+		serializer(ar, waitFailure);
+		if (Archive::isDeserializing) {
+			encryptKeyLookupByKeyId =
+			    RequestStream<struct GetCommitVersionRequest>(waitFailure.getEndpoint().getAdjustedEndpoint(1));
```

Comment:
The adjusted endpoints seems a small bit like overkill for something that's simulation only and only has a couple endpoints, but if it already works it's fine

Reply:
Yeah it works fine. I just wanted to keep it consistent with other interface definitions, no strong opinions, but, given it works we might leave it as is, IMHO.